projects/applicationinsights-angularplugin-js/src/lib/applicationinsights-angularplugin-js.component.ts

```diff
@@ -37,83 +45,130 @@ export class AngularPlugin extends BaseTelemetryPlugin {
     public priority = 186;
     public identifier = 'AngularPlugin';
 
-    private analyticsPlugin: AnalyticsPlugin;
-    private propertiesPlugin: properties.PropertiesPlugin;
-    private _angularCfg: IAngularExtensionConfig = null;
-    private _eventSubscription: Subscription = null;
-
-
     constructor() {
         super();
-        let self = this;
-
-        self._doTeardown = (unloadCtx?: IProcessTelemetryUnloadContext, unloadState?: ITelemetryUnloadState, asyncCallback?: () => void): void | boolean => {
-            if (this.analyticsPlugin && ApplicationinsightsAngularpluginErrorService.instance !== null) {
-                ApplicationinsightsAngularpluginErrorService.instance.plugin = null;
-                if (self._angularCfg) {
-                    if (self._angularCfg.errorServices && Array.isArray(self._angularCfg.errorServices)) {
-                        arrForEach(self._angularCfg.errorServices, (errorService: IErrorService) => {
-                            ApplicationinsightsAngularpluginErrorService.instance.removeErrorHandler(errorService);
-                        });
+        let _analyticsPlugin: AnalyticsPlugin;
+        let _propertiesPlugin: PropertiesPlugin;
+        let _angularCfg: IAngularExtensionConfig;
+        let _eventSubscription: Subscription;
+        let _isPageInitialLoad: boolean;
+        let _prevRouter: Router;
+
+        dynamicProto(AngularPlugin, this, (_self, _base) => {
+            _initDefaults();
+
+            _self.initialize = (config: IConfiguration & IConfig, core: IAppInsightsCore, extensions: IPlugin[], pluginChain?: ITelemetryPluginChain) => {
+                super.initialize(config, core, extensions, pluginChain);
+        
+                _self._addHook(onConfigChange(config, (details) => {
+                    let ctx = _self._getTelCtx();
+                    _angularCfg = ctx.getExtCfg<IAngularExtensionConfig>(_self.identifier, defaultAngularExtensionConfig);
+                    _propertiesPlugin = core.getPlugin<PropertiesPlugin>(PropertiesPluginIdentifier)?.plugin as PropertiesPlugin;
+                    _analyticsPlugin = core.getPlugin<AnalyticsPlugin>(AnalyticsPluginIdentifier)?.plugin as AnalyticsPlugin;
+
+                    if (_analyticsPlugin) {
+                        if (ApplicationinsightsAngularpluginErrorService.instance !== null) {
```

Comment:
Also for minification assign this to a local variable and use it rather than the full name all the time
`ApplicationinsightsAngularpluginErrorService.instance`

This will not only aid in minification , but will also be a slight performance improvement as the runtime doesn't have to lookup and dereference this for every usage.